test_wrap_bad in distributed/elastic/multiprocessing/api_test fails fails when executed with pytest,
e.g. when running via run_test.py.
Skip it.
See https://github.com/pytorch/pytorch/issues/115069

Author: Alexander Grund (TU Dresden)

diff --git a/test/distributed/elastic/multiprocessing/api_test.py b/test/distributed/elastic/multiprocessing/api_test.py
index 6240fbe41d6..8fe6e3bc6d7 100644
--- a/test/distributed/elastic/multiprocessing/api_test.py
+++ b/test/distributed/elastic/multiprocessing/api_test.py
@@ -42,6 +42,7 @@ from torch.testing._internal.common_utils import (
     TestCase,
     run_tests,
     skip_but_pass_in_sandcastle_if,
+    skip_if_pytest,
 )
 
 
@@ -661,6 +662,7 @@ if not (TEST_WITH_DEV_DBG_ASAN or IS_WINDOWS or IS_MACOS):
 if not (TEST_WITH_DEV_DBG_ASAN or IS_WINDOWS or IS_MACOS or IS_CI):
 
     class StartProcessesNotCITest(StartProcessesTest):
+        @skip_if_pytest
         def test_wrap_bad(self):
             none = ""
             stdout_log = os.path.join(self.test_dir, "stdout.log")
diff --git a/torch/_dynamo/testing.py b/torch/_dynamo/testing.py
index 79c869cc481..20605019005 100644
--- a/torch/_dynamo/testing.py
+++ b/torch/_dynamo/testing.py
@@ -36,16 +36,6 @@ def clone_me(x):
     return x.detach().clone().requires_grad_(x.requires_grad)
 
 
-def skip_if_pytest(fn):
-    @functools.wraps(fn)
-    def wrapped(*args, **kwargs):
-        if "PYTEST_CURRENT_TEST" in os.environ:
-            raise unittest.SkipTest("does not work under pytest")
-        return fn(*args, **kwargs)
-
-    return wrapped
-
-
 def named_parameters_for_optimized_module(mod):
     assert isinstance(mod, eval_frame.OptimizedModule)
     return mod._orig_mod.named_parameters
diff --git a/torch/testing/_internal/common_utils.py b/torch/testing/_internal/common_utils.py
index 1e18ca2afec..db2766923ce 100644
--- a/torch/testing/_internal/common_utils.py
+++ b/torch/testing/_internal/common_utils.py
@@ -1541,6 +1541,16 @@ def skipIfTBB(message="This test makes TBB sad"):
     return dec_fn
 
 
+def skip_if_pytest(fn):
+    @wraps(fn)
+    def wrapped(*args, **kwargs):
+        if "PYTEST_CURRENT_TEST" in os.environ:
+            raise unittest.SkipTest("does not work under pytest")
+        return fn(*args, **kwargs)
+
+    return wrapped
+
+
 def slowTest(fn):
     @wraps(fn)
     def wrapper(*args, **kwargs):
